internal/controller/numaflowcontrollerrollout_controller.go

```diff
@@ -154,17 +154,15 @@ func (r *NumaflowControllerRolloutReconciler) Reconcile(ctx context.Context, req
 		return ctrl.Result{}, err
 	}
 
-	deployment, deploymentExists, err := r.getNumaflowControllerDeployment(ctx, numaflowControllerRollout)
+	deployment, _, err := r.getNumaflowControllerDeployment(ctx, numaflowControllerRollout)
 	if err != nil {
 		return ctrl.Result{}, err
 	}
 
-	if deploymentExists {
-		// update our Status with the Deployment's Status
-		err = r.processNumaflowControllerStatus(ctx, numaflowControllerRollout, deployment)
-		if err != nil {
-			return ctrl.Result{}, err
-		}
+	// update our Status with the Deployment's Status
+	err = r.processNumaflowControllerStatus(ctx, numaflowControllerRollout, deployment)
+	if err != nil {
+		return ctrl.Result{}, err
```

Comment:
addresses [this](https://github.com/numaproj/numaplane/pull/125/files#r1697647636) comment